contracts/RewardPool.sol

```diff
@@ -92,6 +96,24 @@ contract RewardPool is AccessControl, ReentrancyGuard {
     uint256 feeAmount,
     uint256 protocolFee
   );
+  event RewardFunctionAddressUpdated(
+    address newRewardFunctionAddress,
+    address previousRewardFunctionAddress
+  );
+  event AddKpi(
+    address indexed user,
+    bytes32 indexed rewardPeriodKey,
+    uint256 amount,
+    uint256 rewardPeriodStart,
+    uint256 rewardPeriodEndExclusive,
+    bytes32 kpiFunctionId
+  );
+  event RewardPeriodProcessed(
+    bytes32 indexed rewardPeriodKey,
+    uint256 rewardPeriodStart,
+    uint256 rewardPeriodEndExclusive,
+    uint256 totalRewardAmount
```

Comment:
Does the reward function always consume that full amount?
Should we add how many users/builders were processed?